src/auth/googleauth.ts

```diff
@@ -322,10 +322,8 @@ export class GoogleAuth<T extends AuthClient = JSONClient> {
       } else if (credential instanceof BaseExternalAccountClient) {
         credential.scopes = this.getAnyScopes();
       }
-      this.cachedCredential = credential;
-      projectId = await this.getProjectIdOptional();
 
-      return {credential, projectId};
+      return this.prepareADCResponse(credential);
```

Comment:
Please do `return await` since `prepareADCResponse` is async. It does not change the behavior in the current code, but the code above does several `await` s so it will be in line with the remaining code. Also, if we ever add `try ... catch` here, `return` of the promise without awaiting will backfire.

Reply:
Sure can add for consistency and future proofing. I skipped adding await because caller will be doing an await anyway.